src/fetchWebSetup.ts

```diff
@@ -42,8 +64,29 @@ export function getCachedWebSetup(options: any): WebConfig | undefined {
  */
 export async function fetchWebSetup(options: any): Promise<WebConfig> {
   const projectId = getProjectId(options, false);
-  const res = await apiClient.get<WebConfig>(`/projects/${projectId}/webApps/-/config`);
+
+  // Try to determine the appId from the default Hosting site, if it is linked.
+  let hostingAppId: string | undefined = undefined;
+  try {
+    const sites = await hostingApiClient.get<ListSitesResponse>(`/projects/${projectId}/sites`);
+    const defaultSite = sites.body.sites.find((s) => s.type === "DEFAULT_SITE");
+    if (defaultSite && defaultSite.appId) {
+      hostingAppId = defaultSite.appId;
+    }
+  } catch (e) {
+    logger.debug("Failed to list hosting sites");
+    logger.debug(e);
+  }
+
+  // Get the web app config for the appId, or use the '-' special value if the appId is not known
+  const appId = hostingAppId || "-";
+  const res = await apiClient.get<WebConfig>(`/projects/${projectId}/webApps/${appId}/config`);
```

Comment:
Probably worth a try/catch around this

Reply:
The only caller has a try/catch that does the right thing.